File: 240328/중앙값 계산 2/get-median-2.java
import java.util.Scanner;
import java.util.Arrays;
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Integer[] arr = new Integer[n];

        for(int i=0;i<n;i++) {
            arr[i] = sc.nextInt();
            if(i%2 == 0) {
                Arrays.sort(arr,0,i+1);
                System.out.printf("%d ", arr[i/2]);
            }
        }

    }
}

File: 240417/중앙값 계산 2/get-median-2.kt
import java.util.Scanner

fun main() {
    val sc = Scanner(System.`in`)

    val n = sc.nextInt()

    val numbers = mutableListOf<Int>()

    repeat(n) {
        val number = sc.nextInt()

        numbers.add(number)

        if (number % 2 != 0) {
            print("${numbers.sorted()[numbers.size / 2]} ")
        }
    }
}